spec/models/publication_spec.rb

```diff
@@ -252,6 +252,46 @@
     end
   end
 
+  describe 'update_from_pubmed' do
+    it 'should not update from pubmed source if there is no pmid' do
+      expect(publication.pmid).to be_nil
+      expect(publication.update_from_pubmed).to be false
+    end
+    it 'should update from pubmed source if there is a pmid' do
+      pmid = 1
+      publication.pmid = pmid
+      source_data = '<PubmedArticle><MedlineCitation Status="Publisher" Owner="NLM"><PMID Version="1">1</PMID><OriginalData/></PubmedArticle>'
+      new_source_data = '<PubmedArticle><MedlineCitation Status="Publisher" Owner="NLM"><PMID Version="1">1</PMID><SomeNewData/></PubmedArticle>'
+      pubmed_record = PubmedSourceRecord.create(pmid: pmid, source_data: source_data)
+      allow(PubmedSourceRecord).to receive(:find_by_pmid).with(pmid).and_return(pubmed_record)
+      expect(pubmed_record.source_data).to be_equivalent_to source_data
+      allow_any_instance_of(PubmedClient).to receive(:fetch_records_for_pmid_list).with(pmid).and_return(new_source_data)
+      expect(publication.pmid).to_not be_nil
+      expect(publication.update_from_pubmed).to be true
+      expect(PubmedSourceRecord.find_by_pmid(pmid).source_data).to be_equivalent_to new_source_data
```

Comment:
Also check that some specific values in the PubMed update make their way into the PubHash.